src/lib/y2firewall/clients/auto.rb

```diff
@@ -62,16 +90,30 @@ def change
         :next
       end
 
+      # Write the imported configuration to firewalld. If for some reason the
+      # configuration was not imported from the profile, it tries to import
+      # it again.
       def write
+        return false if !firewalld.installed?
+        import(self.class.profile) unless self.class.imported
+        return false unless self.class.imported
+
         firewalld.write
+        activate_service
       end
 
+      # Read the currnet firewalld configuration
       def read
-        firewalld.read if firewalld.installed?
+        return false if !firewalld.installed?
+        firewalld.read
       end
 
+      # A map with the packages that needs to be installed or removed for
+      # configuring properly firewalld
+      #
+      # @return packages [Hash] of packages to be installed or removed
```

Comment:
JFYI, you could do `Hash{String => Array<String>}` if I am not wrong (see https://www.yardoc.org/types.html).